addon/components/polaris-button-group.js

```diff
@@ -1,80 +1,67 @@
 import Component from '@ember/component';
-import layout from '../templates/components/polaris-button-group';
-import { wrapChildren, rejectNodesByClassName } from '../utils/dom';
+import { action } from '@ember/object';
+import { tagName, layout } from '@ember-decorators/component';
+import template from '../templates/components/polaris-button-group';
+import AutoWrapper from '../-private/auto-wrapper';
 
 /**
  * Polaris button group component.
  * See https://polaris.shopify.com/components/actions/button-group
  */
-export default Component.extend({
-  classNames: ['Polaris-ButtonGroup'],
-
-  classNameBindings: [
-    'segmented:Polaris-ButtonGroup--segmented',
-    'fullWidth:Polaris-ButtonGroup--fullWidth',
-    'connectedTop:Polaris-ButtonGroup--connectedTop',
-  ],
-
-  layout,
-
+@tagName('')
+@layout(template)
+export default class PolarisButtonGroup extends Component {
   /**
    * Button components
    *
    * This component can be used in block form,
    * in which case the block content will be used
    * instead of `text`
    *
-   * @property text
    * @type {String}
    * @default null
    * @public
    */
-  text: null,
+  text = null;
 
   /**
    * Join buttons as segmented group
    *
-   * @property segmented
    * @type {Boolean}
    * @default false
    * @public
    */
-  segmented: false,
+  segmented = false;
 
   /**
    * Buttons will stretch/shrink to occupy the full width
    *
-   * @property fullWidth
    * @type {Boolean}
    * @default false
    * @public
    */
-  fullWidth: false,
+  fullWidth = false;
 
   /**
    * Remove top left and right border radius
    *
-   * @property connectedTop
    * @type {Boolean}
    * @default false
    * @public
    */
-  connectedTop: false,
-
-  'data-test-button-group': true,
-
-  didRender() {
-    this._super(...arguments);
+  connectedTop = false;
 
-    // Wrap each child element that isn't already a group item.
-    let nodesToWrap = rejectNodesByClassName(
-      this.element.children,
-      'Polaris-ButtonGroup__Item'
-    );
-    let wrapper = document.createElement('div');
+  @action
+  setupAutoWrapper(element) {
+    // TODO When we're auto-wrapping we bypass the group item component, so we will not handle
+    // focus events and hence not apply correct class to the wrapper group item
```

Comment:
Is this a new issue? When does it occur/what are the symptoms?

Also thinking... can we ever handle this without some (probably hacky) workaround? 🤔

Reply:
This is an existing issue....whenever users of this component are not explicitly wrapping buttons inside `buttonGroup.item` components, we will auto-wrap buttons with a `div` (what we do here), but that obviously will not handle focus events that the `buttonGroup.item` component does

Can't think of something nice here, only if we could easily render the item component inline and wrap with it contents....sounds crazy
I'd say it might worth just deprecating this auto-wrapping for components like this (not sure if there are other)

Reply:
Yeah, probably removing auto-wrapping is the way to go to work around this, any other "fix" would probably have to be dirty 😬